ui/src/modals/newtask.vue

```diff
@@ -264,35 +264,54 @@ export default {
             });
 
             //now find apps that user can submit
-            this.$http.get('app', {params: {
-                find: JSON.stringify({
-                    "inputs.datatype": {$in: datatype_ids},
-                    removed: false,
-                }),
+            this.$http.get('/app/query', {params: {
                 sort: 'name', 
                 populate: 'inputs.datatype outputs.datatype',
                 limit: 500, //TODO - this is not sustailable
+                incompatible: true
             }})
             .then(res=>{
                 //now, pick apps that we have *all* input datasets that matches the input datatype/tags
-                res.data.apps.forEach(app=>{
-                    var match = true;
+                res.data.forEach(app=>{
+                    let match = true;
+                    let missingInputIDs = []; // Store missing inputs here
+
                     app.inputs.forEach(input=>{
+                        //In this context, return is used to skip the current iteration and move on to the next one, 
+                        //not to return a value from a function.
```

Comment:
that's fine to remove the comment-- `forEach` never returns, in every context that it is used